### ISSUE TITLE
forEach loops need to be implemented in Fission

### ISSUE THREAD

**Issue by arjunguha** [plasma-umass/Fission#126]

**Comment by arjunguha** [comment]
The interpreter needs to read `ArrayPrototype`. We could change the syntax of `ArrayLit` to propagate it to the right place. 

Here is an alternative approach:

1. Bind `ArrayPrototype` to `undefined` in `initEnv` (in `NodejsCEK`).
2. In `init.quark`, update `ArrayPrototype`. We'll have to change the parser slightly, so that it produces a list of bindings and a body expression.
3. Once `ArrayPrototype` is in `initEnv`, its address can be stored in a module-level variable in `JavaScriptCEKLike`.
4. `ArrayLit` evaluation can read the global variable to setup the prototype correctly.

**Comment by rachitnigam** [comment]
The benchmark BarTinder hits a case where forEach is called.

**Comment by rachitnigam** [comment]
Also the solution that is suggested for ArrayPrototype needs to be implemented for ObjectPrototype too, since currently, VObj creation does not define the `__proto__` on the object as `Object`.

**Comment by rachitnigam** [comment]
@arjunguha Can you add a specific test that needs to pass, but fails right now?

**Comment by rachitnigam** [comment]
This test already passes:
```
var arr = [1, 2, 3, 4];
var sum = 0;
arr.forEach(function(elem) {
  sum += elem;
})
sum
```

When does foreach fail?

**Comment by rachitnigam** [comment]
This fails: 
```
var a = {x: 1};
var a = {x: 1};

**Comment by arjunguha** [comment]
This is because we are using the native forEach function, which forces us to convert to JS. There should be no need to convert to JS.

**Comment by rachitnigam** [comment]
Yeah, I figured that out. I've tried to implement it on the forEach branch.
